builder/vmware/iso/step_create_vmx.go

```diff
@@ -110,14 +367,175 @@ func (s *stepCreateVMX) Run(state multistep.StateBag) multistep.StepAction {
 		}
 	}
 
-	ctx.Data = &vmxTemplateData{
+	templateData := vmxTemplateData{
 		Name:     config.VMName,
 		GuestOS:  config.GuestOSType,
 		DiskName: config.DiskName,
 		Version:  config.Version,
 		ISOPath:  isoPath,
+
+		SCSI_Present:         "FALSE",
+		SCSI_diskAdapterType: "lsilogic",
+		SATA_Present:         "FALSE",
+		NVME_Present:         "FALSE",
+
+		DiskType:              "scsi",
+		CDROMType:             "ide",
+		CDROMType_MasterSlave: "0",
+
+		Sound_Present: map[bool]string{true: "TRUE", false: "FALSE"}[bool(config.Sound)],
+		Usb_Present:   map[bool]string{true: "TRUE", false: "FALSE"}[bool(config.USB)],
+
+		Serial_Present:   "FALSE",
+		Parallel_Present: "FALSE",
+	}
+
+	/// Use the disk adapter type that the user specified to tweak the .vmx
+	//  Also sync the cdrom adapter type according to what's common for that disk type.
+	diskAdapterType := strings.ToLower(config.DiskAdapterType)
+	switch diskAdapterType {
+	case "ide":
+		templateData.DiskType = "ide"
+		templateData.CDROMType = "ide"
+		templateData.CDROMType_MasterSlave = "1"
+	case "sata":
+		templateData.SATA_Present = "TRUE"
+		templateData.DiskType = "sata"
+		templateData.CDROMType = "sata"
+		templateData.CDROMType_MasterSlave = "1"
+	case "nvme":
+		templateData.NVME_Present = "TRUE"
+		templateData.DiskType = "nvme"
+		templateData.SATA_Present = "TRUE"
+		templateData.CDROMType = "sata"
+		templateData.CDROMType_MasterSlave = "0"
+	case "scsi":
+		diskAdapterType = "lsilogic"
+		fallthrough
+	default:
+		templateData.SCSI_Present = "TRUE"
+		templateData.SCSI_diskAdapterType = diskAdapterType
+		templateData.DiskType = "scsi"
+		templateData.CDROMType = "ide"
```

Comment:
When using SCSI/lsilogic for the disk_adapter_type I think it makes more sense here to have the CDRomType also become SCSI, right?

Reply:
I suppose so. I just duplicated the state that VMware Workstation put the .vmx in when I build VMs using those settings. By default VMware chose IDE for the cdrom when SCSI was chosen for the disk. So I just chose that to not do anything that VMware Workstation didn't expect.

Reply:
Hrm, interesting. I got SCSI CDRom, but I'm building with virtualHW.version = "7" I think, so I guess the rules are more complex. I needed SCSI in this case, hence the ask :-)
Would it make sense just to expose CDRomType as cdrom_adapter_type?